src/main/java/net/sf/jabref/logic/util/Version.java

```diff
@@ -5,63 +5,93 @@
 import java.io.InputStreamReader;
 import java.net.URL;
 import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
 import org.json.JSONObject;
 
 /**
  * Represents the Application Version with the major and minor number, the full Version String and if it's a developer version
  */
 public class Version {
 
-    public static final String JABREF_DOWNLOAD_URL = "https://downloads.jabref.org";
     private static final Log LOGGER = LogFactory.getLog(Version.class);
-    private static final String JABREF_GITHUB_URL = "https://api.github.com/repos/JabRef/jabref/releases/latest";
+
+    private final static Pattern VERSION_PATTERN = Pattern.compile("(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(?<stage>-alpha|-beta)?(?<dev>-?dev)?.*");
+
+    public static final String JABREF_DOWNLOAD_URL = "https://downloads.jabref.org";
+    private static final String JABREF_GITHUB_RELEASES = "https://api.github.com/repos/JabRef/JabRef/releases";
+
 
     private String fullVersion = BuildInfo.UNKNOWN_VERSION;
-    private int major;
-    private int minor;
-    private int patch;
+    private int major = -1;
+    private int minor = -1;
+    private int patch = -1;
+    private DevelopmentStage developmentStage = DevelopmentStage.UNKNOWN;
     private boolean isDevelopmentVersion;
 
     /**
-     * @param version must be in form of X.X (e.g., 3.3; 3.4dev)
+     * @param version must be in form of following pattern: {@code (\d+)(\.(\d+))?(\.(\d+))?(-alpha|-beta)?(-?dev)?} (e.g., 3.3; 3.4-dev)
      */
     public Version(String version) {
         if ((version == null) || "".equals(version) || version.equals(BuildInfo.UNKNOWN_VERSION)
                 || "${version}".equals(version)) {
             return;
         }
 
-        String[] versionParts = version.split("dev");
-        String[] versionNumbers = versionParts[0].split(Pattern.quote("."));
-        try {
-            this.major = Integer.parseInt(versionNumbers[0]);
-            this.minor = versionNumbers.length >= 2 ? Integer.parseInt(versionNumbers[1]) : 0;
-            this.patch = versionNumbers.length >= 3 ? Integer.parseInt(versionNumbers[2]) : 0;
-            this.fullVersion = version;
-            this.isDevelopmentVersion = version.contains("dev");
-        } catch (NumberFormatException exception) {
-            LOGGER.warn("Invalid version string used: " + version, exception);
+        this.fullVersion = version;
```

Comment:
Could you please extract the parsing logic to a new `Version parse(String version)` method. The constructor should only accept already-parsed values (i.e. major, minor etc as integers and the developmentStage). In this way the Version class adheres to our conventions for parsing from string values.